include/openthread/dnssd_server.h

```diff
@@ -147,6 +147,30 @@ typedef enum
     OT_DNSSD_QUERY_TYPE_RESOLVE_HOST = 3, ///< Service type resolve hostname.
 } otDnssdQueryType;
 
+/**
+ * Includes the failure counters due to different reasons.
+ *
+ */
+typedef struct otUpstreamDnsFailureCounters
+{
+    uint32_t mTotalFailure; ///< Sum up of all failures.
```

Comment:
Is it necessary to have this field? Can we just sum up all other failure type counters?

Reply:
My concern is this field needs more maintenance. I.e., when you meet an error you need to update two counters: the `mTotalFailure` and the specific error type counter.